### PR TITLE
refactor(msg-handling): removing duplicated msg signature verification

### PR DESCRIPTION
**Pull request by bochaco**
Also re-organising msg handing moving certain validations into the corresponding msg type handler.

### CODE REVIEW

File: src/routing/core/msg_handling/relocation.rs [hunk 1]
```diff
@@ -138,6 +138,24 @@ impl Core {
...
+                // Already received a promise or already relocating. discard.
+                return Ok(vec![]);
+            }
+        } else {
+            // Promise returned from a node to be relocated, to be exchanged for the actual
+            // `Relocate` message.
+            if self.is_not_elder() || self.section.is_elder(&promise.name) {
+                // If we are not elder, maybe we just haven't processed our promotion yet.
+                // If they are still elder, maybe we just haven't processed their demotion yet.
+                // If otherwise they are still elder, maybe we just haven't processed their demotion yet.
```
Comment (maqi) [inline src/routing/core/msg_handling/relocation.rs:154]: this line is duplicated with the above?

File: src/routing/routing_api/tests/mod.rs [hunk 2]
```diff
@@ -1585,7 +1585,9 @@ async fn handle_elders_update() -> Result<()> {
         assert_eq!(section.chain().last_key(), &pk1);
 
         // The message is trusted even by peers who don't yet know the new section key.
-        assert_matches!(message.verify(iter::once(&pk0)), Ok(VerifyStatus::Full));
+        // FIXME
+        //assert!(message.verify(iter::once(&pk0)));
```
Comment (maqi) [inline src/routing/routing_api/tests/mod.rs:1589]: This assertion is now failing?

Comment (maqi) [inline src/routing/routing_api/tests/mod.rs:1589]: This assertion is now failing?

File: src/routing/core/msg_handling/mod.rs [hunk 3]
```diff
@@ -65,111 +62,73 @@ impl Core {
...
 
-        Ok(commands)
-    }
-
-    pub(crate) fn handle_timeout(&mut self, token: u64) -> Result<Vec<Command>> {
-        self.dkg_voter
-            .handle_timeout(&self.node.keypair, token)
-            .into_commands(&self.node, *self.section_chain().last_key())
-    }
+        trace!("Useful message from {:?}: {:?}", sender, msg);
```
Comment (maqi) [inline src/routing/core/msg_handling/mod.rs:81]: The above check only ensures the message is `trusted` , not `useful` ?

File: src/routing/routing_api/tests/mod.rs (hunk 2 above)
Comment (bochaco) [reply src/routing/routing_api/tests/mod.rs:1589]: The `verify` was changed and it doesn't perform all the checks it used to do before, so I need to replace it with a couple of calls here to have the test verify the same as before

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**